lib/provider.py

```diff
@@ -107,26 +127,41 @@ def perform_search(search_type, data):
         logging.warning("No scrapers configured/enabled")
         return None
 
+    if not isinstance(search_types, list):
+        search_types = [search_types]
+
     runner_class = ProgressScraperRunner if get_boolean_setting("enable_bg_dialog") else ScraperRunner
     with runner_class(scrapers, num_threads=get_int_setting("thread_number")) as runner:
-        runner_data = runner.parse_query(data) if search_type == "query" else runner.parse(search_type, data)
-
-        for scraper, scraper_results in runner_data:
-            logging.debug("Processing %s scraper results", scraper.name)
-            for scraper_result in scraper_results:
-                magnet = Magnet(scraper_result["magnet"])
-                try:
-                    info_hash = magnet.parse_info_hash()
-                except InvalidMagnet:
-                    continue
-                if info_hash == "0" * 40:
-                    continue
-
-                magnet_result = results.get(info_hash)  # type: Result
-                if magnet_result is None:
-                    results[info_hash] = Result(scraper, scraper_result)
-                else:
-                    magnet_result.add_result(scraper, scraper_result)
+        for search_type in search_types:
+            runner_data = runner.parse_query(data) if search_type == "query" else runner.parse(search_type, data)
+
+            for scraper, scraper_results in runner_data:
+                logging.debug("Processing %s scraper results", scraper.name)
+                for scraper_result in scraper_results:
+                    title = scraper_result["title"]
+                    if not title:
+                        continue
+
+                    if "episode" in search_types and search_type == "season":
```

Comment:
When including seasons in episode search, if the result does not match `Exx-xx` and contains `Exx` only once it does not include the result because it probably is a single episode. Any suggestions are welcome.

Reply:
What confuses me here is the use of Exx. Shouldn't it be Sxx?
Anyway I think we will skip this part for now until we find a better solution.